<!DOCTYPE html>
<html>
<head>
<title>03_QualityAndAssembly.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>
<link rel="stylesheet" href="file:///home/s.varona/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>
<link rel="stylesheet" href="file:///home/s.varona/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="bacterial-wgs-training--exercise-3">Bacterial WGS training : Exercise 3</h1>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>Sequence quality and assambly.</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td></td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>How do I know if my data was correctly sequenced?</li><li>How can I improve my data quality?</li><li>How do I assamble the reads?</li><li>How do I know if my reads were correctly assembled</li></ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Check quality of sequenced data.</li><li>Trimm low quality segments and adapters.</li><li>Assamble mapped reads.</li></ul></td>
</tr>
<tr>
<td><strong>Time estimation</strong>:</td>
<td>1 h 30 min</td>
</tr>
<tr>
<td><strong>Key points</strong>:</td>
<td><ul><li>Analysis of sequence quality.</li><li>Assembly.</li></ul></td>
</tr>
</tbody>
</table>
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#exercise">Exercise</a>
<ul>
<li><a href="#preprocessing">Preprocessing</a></li>
<li><a href="#assembly">Assembly</a></li>
</ul>
</li>
</ul>
<div class="tables-end"></div>
<h2 id="introduction">Introduction</h2>
<h3 id="training-summary">Training summary</h3>
<p align="center"><img src="img/bacterial_wgs_training.png" alt="Fastqc_1" width="500"></p>
<h3 id="training-dataset-description">Training dataset description</h3>
<p>This dataset was used for <a href="https://ecdc.europa.eu/en/publications-data/fifth-external-quality-assessment-scheme-listeria-monocytogenes-typing">external quality assessment (EQA-5)</a> scheme for typing of
Listeria monocytogenes (<em>L. monocytogenes</em>) organised for laboratories providing data to the Food and Waterborne
Diseases and Zoonoses Network (FWD-Net) managed by ECDC. Since 2012, the Section for Foodborne Infections
at the Statens Serum Institut (SSI) in Denmark has arranged this EQA under a framework contract with ECDC. The
EQA-5 contain serotyping and molecular typing-based cluster analysis.</p>
<p>Human listeriosis is a relatively rare but serious zoonotic disease with an EU notification rate of 0.47 cases per 100
000 population in 2016. The number of human listeriosis cases in the EU has increased since 2008, with the
highest annual number of deaths since 2009 reported in 2015 at 270.</p>
<p>The objectives of the EQA are to assess the quality and comparability of the typing data reported by public health
national reference laboratories participating in FWD-Net. Test isolates for the EQA were selected to cover isolates
currently relevant to public health in Europe and represent a broad range of clinically relevant types for invasive
listeriosis. Two separate sets of 11 test isolates were selected for serotyping and molecular typing-based cluster
analysis. The expected cluster was based on a pre-defined categorisation by the organiser.
Twenty-two <em>L. monocytogenes</em> test isolates were selected to fulfil the following criteria:</p>
<ul>
<li>cover a broad range of the common clinically relevant types for invasive listeriosis</li>
<li>include closely related isolates</li>
<li>remain stable during the preliminary test period at the organising laboratory.</li>
</ul>
<p>The 11 test isolates for cluster analysis were selected to include isolates with different or varying relatedness isolates and different multi locus sequence types.</p>
<h3 id="how-do-i-know-if-my-data-was-correctly-sequenced">How do I know if my data was correctly sequenced?</h3>
<p>Despite the improvement of sequencing methods, there is no error-free technique. The Phred quality score <a href="https://www.ncbi.nlm.nih.gov/pubmed/9521921">(Ewing et al., 1998)</a> has been used since the late 90s as a measure of the quality of each sequenced nucleotide. Phred quality scores not only allow us to determine the accuracy of sequencing and of each individual position in an assembled consensus sequence, but it is also used to compare the efficiency of the sequencing methods.</p>
<p>Phred quality scores Q are defined as a property which is logarithmically related to the base-calling error probabilities P . The Phred quality score is the negative ratio of the error probability to the reference level of P = 1 expressed in Decibel (dB):</p>
<p align="center">ùëÑ = ‚àí10 log<sub>10</sub> P</p>
<p>A correct measuring of the sequencing quality is essential for identifying problems in the sequencing and removal of low-quality sequences or sub sequences. Conversion of typical Phred scores used for quality thresholds into accuracy can be ead in the following table:</p>
<table>
<thead>
<tr>
<th><strong>Phred score</strong></th>
<th>Error probability</th>
<th>Accuracy</th>
</tr>
</thead>
<tbody>
<tr>
<td>10</td>
<td>1/10</td>
<td>90%</td>
</tr>
<tr>
<td>20</td>
<td>1/100</td>
<td>99%</td>
</tr>
<tr>
<td>30</td>
<td>1/1000</td>
<td>99.9%</td>
</tr>
<tr>
<td>40</td>
<td>1/10000</td>
<td>99.99%</td>
</tr>
</tbody>
</table>
<p>There are multiple software to read and generate statistics to help with the interpretation of the quality of a sequence. One of the most commonly used methods for this task is <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (Andrews, 2010), a java program that run on any system and has both command line and graphic interface.</p>
<p align="center"><img src="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc.png" alt="Fastqc_1" width="500"></p>
<br>
<p>FastQC aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing. It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis. Here you can compare examples of a <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good sequencing output</a> and a <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">bad one</a>.</p>
<h3 id="how-can-i-improve-my-data-quality">How can I improve my data quality?</h3>
<p>Most modern aligners can filter out low quality reads and clip off low quality ends and adapters. In case it has to be done manually, because the sequencing was poor but you still need to use that data or because you want to have more control on the trimming of reads or use a particular method, there are standalone applications that allow you to do it. <a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a> (Bolger, Lohse, &amp; Usadel, 2014) is one of the most broadly used along with <a href="https://bioinformaticshome.com/tools/rna-seq/descriptions/fastp.html">fastp</a> (Chen S. et al, 2018). Fastp is the one we will use in the training . It is written in C++, has multi-threading support and performs a variety of useful trimming tasks for illumina paired-end and single ended data.</p>
<p>Common trimming includes removal of short reads, and cut off adapters and a number of bases, if below a threshold quality. Modern algorithms also include more complex methods, as the sliding window trimming. This is the method we will use in the exercises, and it allows to trimm a variable number of bases in each read, cutting once the average quality within the window falls below a threshold.</p>
<h3 id="de-novo-or-reference-based-assembly">De novo or reference-based assembly?</h3>
<p>After preprocessing, the next step is aligning the reads to rebuild the genomic sequence. There are two main ways of doing this:</p>
<ul>
<li>
<p>Reference-based assembly</p>
<p>For each of the short reads in the FASTQ file, a corresponding location in the reference sequence is determined. A mapping algorithm will locate a location in the reference sequence that matches the read, while tolerating a certain amount of mismatch to allow subsequence variation detection tath correspond to the actual difference between the reference and de assembled genome.</p>
</li>
<li>
<p><em>De novo</em> assembly</p>
<p><em>De novo</em> genome assembly consists in taking a collection of short sequencing reads and reconstruct the genome sequence, source of all these fragments.The output of an assembler is decomposed into contigs: contiguous regions of the genome which are resolved, and/or scaffolds: longer sequences formed by reordered and oriented contigs with positional information but without sequence resolution.</p>
</li>
</ul>
<h2 id="exercise">Exercise</h2>
<h3 id="preprocessing">Preprocessing</h3>
<p>As we have seen in the introduction, the first step is to know the quality of our sequences. Those with an unnaceptable quality will be trimmed in order to remove the nucleotides with bad quality to ease future analysis algorithms such assembly.
In order to check quality and trim the reads wee have to execute this command:</p>
<hr>
<pre class="hljs"><code><div><span class="hljs-built_in">cd</span>
<span class="hljs-built_in">pwd</span>
<span class="hljs-comment">#Output: /home/alumno</span>
<span class="hljs-built_in">cd</span> wgs/bacterial_wgs_training_dataset/ANALYSIS/
<span class="hljs-built_in">pwd</span>
<span class="hljs-comment">#Output: /home/alumno/wgs/bacterial_wgs_training_dataset/ANALYSIS/</span>
ls
<span class="hljs-comment">#Output: 01-handsonlinux</span>
</div></code></pre>
<p>Now we only have the folder from the first day of training. And here in <code>ANALYSIS</code> folder is where we will run the nextflow.
First we activate the nextflow conda environment</p>
<pre class="hljs"><code><div>micromamba deactivate
micromamba activate nextflow
</div></code></pre>
<p>An now run the main nextflow command for pre-processing</p>
<pre class="hljs"><code><div>nextflow run ../../bacterial_wgs_training/main.nf \
  --reads <span class="hljs-string">'../RAW/DOWNSAMPLED/*_R{1,2}.fastq.gz'</span> \
  -profile conda \
  --fasta ../REFERENCES/listeria_NC_021827.1_NoPhagues.fna \
  --step preprocessing  \
  --outdir 01-preprocessing
</div></code></pre>
<hr>
<p>This execution runs internally three programs: FastQC, fastp and MultiQC as follow:</p>
<p>For each sample those command are executed:</p>
<ul>
<li>
<p><code>fastqc reads_R1.fastq.gz reads_R2.fastq.gz</code></p>
<ul>
<li>reads_R1.fastq.gz and reads_R2.fastq.gz are the input Illumina reads which quality is analyzed</li>
</ul>
</li>
<li>
<p><code>fastp --in1 reads_R1.fastq.gz --in1 reads_R2.fastq.gz --out1 reads_trimmed_R1.fastq --out2 reads_trimmed_R2.fastq --unpaired1 reads_unpaired_R1.fastq --unpaired2 reads_unpaired_R2.fastq --detect_adapter_for_pe --cut_front --cut_tail --cut_mean_quality 20 --qualified_quality_phred 20 --unqualified_percent_limit 10 --length_required 50 --trim_poly_x --thread 1 --json sample.fastp.json --html sample.fastp.html 2&gt; sample.fastp.log</code></p>
<ul>
<li>reads_R1.fastq.gz and reads_R2.fastq.gz are the input Illumina reads which will be trimmed</li>
<li>reads_trimmed_R[1|2].fastq reads_fail_R[1|2].fastq
<ul>
<li>trimmed refer to sequences trimmed that passed the quality filter for both R1 and R2</li>
<li>fail refer to sequences that did not pass the quality filter</li>
</ul>
</li>
<li>detect_adapter_for_pe: Detects automatically adapters for pair-end data</li>
<li>cut_front and cut_tail: remove bases with low quality from 5' to tail and 3' to front respectively using a sliding window.</li>
<li>cut_mean_quality: mean quality requirement for cut_front and cut_tail. The bases in the sliding window with mean quality below 20 will be cut.</li>
<li>qualified_quality_phred: The quality value that a base is qualified. Bases below 20Q Phred are unqualified.</li>
<li>unqualified_percent_limit: How many percents of bases are allowed to be unqualified. Maximum 10% of bases can be unqualified.</li>
<li>length_required: reads shorter than 50 will be discarded</li>
<li>trim_poly_x: enable polyX trimming in 3' ends.</li>
</ul>
</li>
<li>
<p><code>fastqc reads_trimmed_R[1|2].fastq reads_fail_R[1|2].fastq</code></p>
<ul>
<li>reads_trimmed_R[1|2].fastq reads_fail_R[1|2].fastq are sequences after trimming step which quality will be assesed</li>
</ul>
</li>
<li>
<p><code>multiqc RESULTS_DIRECTORY</code></p>
<ul>
<li>MultiQC will automatically search for raw and trimmed reads quality results and will compare them in user-friendly graphs</li>
</ul>
</li>
</ul>
<h4 id="final-results-should-look-like-those">Final results should look like those</h4>
<table>
<thead>
<tr>
<th>Before trimming</th>
<th>After trimming</th>
</tr>
</thead>
<tbody>
<tr>
<td><img src="img/Ex_2_1.png" alt=""></td>
<td><img src="img/Ex_2_02.png" alt=""></td>
</tr>
</tbody>
</table>
<p>Here we can see the quality of the R1 reads before and after trimming</p>
<p align="center"><img src="img/Ex_2_03.png" alt="Fastqc_4" width="900"></p>
This is the MultiQC output comparing the quality of trimmed and raw reads
<br>
<br>
<!---
<p align="center"><img src="img/Ex_2_4.png" alt="Trimmomatic_1" width="900"></p>
This is the MultiQC output sumarizing reads that have been filtered after trimming step
<br>
<br>-->
<h3 id="assembly">Assembly</h3>
<p>Reconstruct the source genome is a mandatory step for latter analysis such annotation, comparative analysis and cgMLST. In order to assemble all samples we need to run this command:</p>
<hr>
<p>First check that you are in the correct folder:</p>
<pre class="hljs"><code><div><span class="hljs-built_in">pwd</span>
<span class="hljs-built_in">cd</span>
<span class="hljs-built_in">pwd</span>
<span class="hljs-comment">#Output: /home/alumno</span>
<span class="hljs-built_in">cd</span> wgs/bacterial_wgs_training_dataset/ANALYSIS/
<span class="hljs-built_in">pwd</span>
<span class="hljs-comment">#Output: /home/alumno/wgs/bacterial_wgs_training_dataset/ANALYSIS/</span>
ls
<span class="hljs-comment">#Output: 01-handsonlinux 02-assembly work</span>
</div></code></pre>
<p>Now we can run the assembly process</p>
<pre class="hljs"><code><div>nextflow run ../../bacterial_wgs_training/main.nf \
  -resume \
  -profile conda \
  --reads <span class="hljs-string">'../RAW/DOWNSAMPLED/*_R{1,2}.fastq.gz'</span> \
  --fasta ../REFERENCES/listeria_NC_021827.1_NoPhagues.fna \
  --gtf ../REFERENCES/listeria_NC_021827.1_NoPhagues.gff \
  --step assembly \
  --outdir 02-assembly
</div></code></pre>
<hr>
<p>This execution runs internally four programs: FastQC, fastp, <strong>Unicycler</strong> , MultiQC and Quast:</p>
<p>Software for preprocessing are executed as noted before. This step includes an aditional assembly process that uses <a href="https://github.com/rrwick/Unicycler">Unicycler</a> to assemble all samples. Nextflow runs Unicycler for each sample as follow:</p>
<p><code>unicycler -1 reads_R1.fastq.gz -2 reads_R2.fastq.gz</code></p>
<p>Once assembled, the file containing the contigs (SAMPLE_paired_assembly.fasta) will be checked by <a href="http://quast.sourceforge.net/quast">Quast</a>, using a <a href="https://www.ncbi.nlm.nih.gov/nuccore/NC_021827">reference fasta and gff</a> from <em>Listeria monocytogenes</em> strain J1817. Quast was executed as follow:</p>
<p><code>quast.py -R reference.fasta -G reference_genes.gff *_paired_assembly.fasta</code></p>
<ul>
<li>R is the fasta file with the reference sequence</li>
<li>G refers to the gff file for the same sequence</li>
<li>All sequences are used as input with a wildcard that includes all fasta files with assembled contigs</li>
</ul>
<h4 id="final-results-should-look-like-these">Final results should look like these</h4>
<p align="center"><img src="img/Ex_2_6.png" alt="Trimmomatic_1" width="900"></p>
This is the MultiQC output sumarizing the number of contigs assembled for each sample, sorting them by size
<br>
<br>
<p align="center"><img src="img/Ex_2_5.png" alt="Trimmomatic_1" width="900"></p>
This is the MultiQC output sumarizing basic assembly statistics for each sample
<br>
<br>
<p align="center"><img src="img/Ex_2_7.png" alt="Trimmomatic_1" width="900"></p>
This is the Quast output sumarizing all detailed assembly statistics, comparing all the assemblies in one table
<br>
<br>
<p align="center"><img src="img/Ex_2_8.png" alt="Trimmomatic_1" width="900"></p>
This is Quast Icarus viewer where contigs are aligned to the reference supplied in order to check how similar the assembled contigs are to the genome from the database.
<br>
<br>

</body>
</html>
