<!DOCTYPE html>
<html>
<head>
<title>05_outbreakcgMLST.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>
<link rel="stylesheet" href="file:///home/s.varona/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>
<link rel="stylesheet" href="file:///home/s.varona/C%3A%5CUsers%5CSara%5CGoogle%20Drive%5Coposiciones%5Copos_tecnologo%5Coposiciones%5Ccss%5Ccustom_md_style.css" type="text/css">
<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="bacterial-wgs-training--exercise-5">Bacterial WGS training : Exercise 5</h1>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th><strong>Title</strong></th>
<th>cgMLST bacterial outbreak investigation.</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Training dataset:</strong></td>
<td></td>
</tr>
<tr>
<td><strong>Questions:</strong></td>
<td><ul><li>Do I have the needed depth of coverage?</li><li>Do I have correct assemblies?</li><li>How do I download a cgMLST schema?</li><li>How can I analyze my samples using a cgMLST schema?</li><li>How do I visualize the results?</li><li>Which strains belong to the outbreak?</li></ul></td>
</tr>
<tr>
<td><strong>Objectives</strong>:</td>
<td><ul><li>Trimming and quality control of raw reads.</li><li>Assembly and quality control</li><li>cgMLST analysis</li><li>Minnimum spanning tree visualization</li><li>Results interpretation</li></ul></td>
</tr>
<tr>
<td><strong>Time estimation</strong>:</td>
<td>1 h</td>
</tr>
<tr>
<td><strong>Key points</strong>:</td>
<td><ul><li>Importance of assembly in cgMLST typification.</li><li>Summary of alleles reconstruction, and missing data is important.</li><li>Interpretation of results is case, species and epidemiology dependant.</li></ul></td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<h2 id="introduction">Introduction</h2>
<p align="center"><img src="img/taranis_diagram_identify_alleles.jpg" width="1000"></p>
<h2 id="preprocessing">Preprocessing</h2>
<p>Addressed in previous exercises.</p>
<h2 id="strain-characterization-serogroup-and-sequence-type-st-determination-using-wgs-data">Strain characterization: Serogroup and sequence type (ST) determination using WGS data.</h2>
<p>Performing MLST, serogroup or resistance analysis can't be easing using WGS. Here we are going to use a mapping approach using <a href="https://github.com/katholt/srst2">srst2</a> software for determining ST and serogroup of the listeria isolates without doing any PCR.</p>
<h3 id="run-the-exercise">Run the exercise</h3>
<pre class="hljs"><code><div><span class="hljs-built_in">cd</span>
<span class="hljs-built_in">cd</span> wgs/bacterial_wgs_training_dataset/ANALYSIS
nextflow run ../../bacterial_wgs_training/main.nf \
      --reads <span class="hljs-string">'../RAW/DOWNSAMPLED/*_R{1,2}*.fastq.gz'</span> \
      --fasta ../REFERENCES/listeria_NC_021827.1_NoPhagues.fna \
      -profile conda \
      --step strainCharacterization \
      --srst2_db_mlst ../REFERENCES/mlst_pasteur_listeria.fas \
      --srst2_def_mlst ../REFERENCES/mlst_pasteur_listeria.scheme \
      --srst2_db_sero ../REFERENCES/pcr_serogroup_listeria.fas \
      --srst2_def_sero ../REFERENCES/pcr_serogroup_listeria.scheme \
      --srst2_resistance ../REFERENCES/ARGannot.r1.fasta \
      --outdir 05-strainCharacterization \
      -resume
</div></code></pre>
<p>Below this command two srst2 commands are performed using the mlst and serogroup schema downloaded from <a href="http://bigsdb.pasteur.fr/perl/bigsdb/bigsdb.pl?db=pubmlst_listeria_seqdef_public">Pasteur bigsdb</a>.</p>
<pre class="hljs"><code><div>srst2 --input_pe reads_R1.fastq.gz readsR2 \
--forward <span class="hljs-string">"_R1"</span> --reverse <span class="hljs-string">"_R2"</span> --output output \
--<span class="hljs-built_in">log</span> --mlst_db db_mlst --mlst_definition mlst_scheme \
--mlst_delimiter <span class="hljs-string">"_"</span>
</div></code></pre>
<p>Parameters:</p>
<ul>
<li>input_pe: fastq reads</li>
<li>forward/reverse: name of R1-R2 reads for file name parsing.</li>
<li>output: how to name the results.</li>
<li>mlst_db: fasta file with all the alleles for all the genes present in the schema.</li>
<li>mlst_definition: plain text file with the definition of the STs with which alleles. This is known as the mlst profile.</li>
</ul>
<blockquote>
<p>mlst_db and mlst_definition both can be downloaded from Pasteur bigsdb of PubMLST site for a bunch of different microorganisms.</p>
</blockquote>
<h3 id="results-analysis">Results analysis</h3>
<p>The serogroup results look like this, and it can be found in the next path:<code>/home/alumno/course_shared_folder/results/SRST2_SERO/summary.txt</code></p>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th>Sample</th>
<th>ST</th>
<th>lmo0737</th>
<th>lmo1118</th>
<th>ORF2110</th>
<th>ORF2819</th>
<th>prs</th>
<th>mismatches</th>
<th>depth</th>
<th>maxMAF</th>
</tr>
</thead>
<tbody>
<tr>
<td>RA-L2073</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>37.62</td>
<td>0.06</td>
</tr>
<tr>
<td>RA-L2281</td>
<td>IVb (2)</td>
<td>-</td>
<td>-</td>
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>50.97</td>
<td>0.05</td>
</tr>
<tr>
<td>RA-L2327</td>
<td>IVb (2)</td>
<td>-</td>
<td>-</td>
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>45.42</td>
<td>0.06</td>
</tr>
<tr>
<td>RA-L2391</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>45.89</td>
<td>0.058</td>
</tr>
<tr>
<td>RA-L2450</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>50.91</td>
<td>0.086</td>
</tr>
<tr>
<td>RA-L2677</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>56.10</td>
<td>0.060</td>
</tr>
<tr>
<td>RA-L2701</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>54.28</td>
<td>0.066</td>
</tr>
<tr>
<td>RA-L2709</td>
<td>NF</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td>RA-L2782</td>
<td>IVb-v1 (1)</td>
<td>7</td>
<td>-</td>
<td>1</td>
<td>1</td>
<td>2</td>
<td>0</td>
<td>62.38</td>
<td>0.03</td>
</tr>
<tr>
<td>RA-L2805</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>49.67</td>
<td>0.06</td>
</tr>
<tr>
<td>RA-L2978</td>
<td>IVb (13)</td>
<td>-</td>
<td>-</td>
<td>3</td>
<td>3</td>
<td>2</td>
<td>0</td>
<td>50.19</td>
<td>0.05</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<p>The MLST results look like this other table, and can be found in:<code>/home/alumno/course_shared_folder/results/SRST2_MLST/summary.txt</code></p>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th>Sample</th>
<th>ST</th>
<th>abcZ</th>
<th>bglA</th>
<th>cat</th>
<th>dapE</th>
<th>dat</th>
<th>ldh</th>
<th>lhkA</th>
<th>mismatches</th>
<th>depth</th>
<th>maxMAF</th>
</tr>
</thead>
<tbody>
<tr>
<td>RA-L2073</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>59.05</td>
<td>0.063</td>
</tr>
<tr>
<td>RA-L2281</td>
<td>1</td>
<td>3</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>3</td>
<td>1</td>
<td>3</td>
<td>0</td>
<td>62.81</td>
<td>0.05</td>
</tr>
<tr>
<td>RA-L2327</td>
<td>213</td>
<td>1</td>
<td>1</td>
<td>9</td>
<td>13</td>
<td>2</td>
<td>5</td>
<td>5</td>
<td>0</td>
<td>53.07</td>
<td>0.06</td>
</tr>
<tr>
<td>RA-L2391</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>55.90</td>
<td>0.04</td>
</tr>
<tr>
<td>RA-L2450</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>54.76</td>
<td>0.05</td>
</tr>
<tr>
<td>RA-L2677</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>62.47</td>
<td>0.04</td>
</tr>
<tr>
<td>RA-L2701</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>67.59</td>
<td>0.03</td>
</tr>
<tr>
<td>RA-L2709</td>
<td>failed</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td>RA-L2782</td>
<td>382</td>
<td>1</td>
<td>51</td>
<td>11</td>
<td>13</td>
<td>2</td>
<td>5</td>
<td>5</td>
<td>0</td>
<td>55.89</td>
<td>0.05</td>
</tr>
<tr>
<td>RA-L2805</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>66.46</td>
<td>0.04</td>
</tr>
<tr>
<td>RA-L2978</td>
<td>6</td>
<td>3</td>
<td>9</td>
<td>9</td>
<td>3</td>
<td>3</td>
<td>1</td>
<td>5</td>
<td>0</td>
<td>52.40</td>
<td>0.06</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<p>We will describe here the meaning of each column:</p>
<ul>
<li>Sample: sample name</li>
<li>ST: serotype or serogroup determined.</li>
<li>lmo0737-prs: names of the genes present in the MLST schema in this case. This column will vary depending on the species and the schema used. Each column shows the allele number determined for each sample.</li>
<li>mismatches: number of mismatches (SNPs) found against the reference allele.</li>
<li>uncertainty: a score showing the probability of having determined the correct allele (ST).</li>
<li>depth: depth of coverage achieved mapping against this allele.</li>
<li>maxMAF: maximum Minimum Allele frequency, this shows the percentage of the samples having the same allele that this sample.</li>
</ul>
<p>And finally we can plot a clustering using MLST profile with a resistance heatmap for visualization:</p>
<p align="center"><img src="img/mlst_resistance.png" width="1000"></p>
<h2 id="assembly">Assembly</h2>
<p>Addressed in previous exercises.</p>
<h2 id="cgmlst-analysis">cgMLST Analysis</h2>
<p>We are using Taranis as the main software for cgMLST analysis. Following the development of the former exercises we are using nextflow, in this case using <code>outbreakMLST</code> step.
This step includes the following processes:</p>
<ul>
<li>Preprocessing with trimmomatic and FastQC.</li>
<li>Assembly and quality control.</li>
<li>Download of cgMLST schema for L. monocytogenes from Pasteur bigsdb site.</li>
<li>cgMLST analysis using Taranis app.</li>
</ul>
<h3 id="run-the-exercise">Run the exercise</h3>
<p>First of all we need to be clear in which folder we are. We need to be in our working directory <code>/home/alumno/Documents/wgs</code> and our training dataset downloaded the first day must be there (If you had any problem the previous sessions please refere to the <a href="00_SetUp.md">setup tutorial</a>).</p>
<p>You can run this command to check where you are:</p>
<pre class="hljs"><code><div><span class="hljs-built_in">pwd</span>
</div></code></pre>
<p>Output:</p>
<pre class="hljs"><code><div>/home/alumno/wgs/bacterial_wgs_training_dataset/ANALYSIS
</div></code></pre>
<p>And this one to list all the files in your working directory. Check there is the training_dataset folder and the results folder from previous sessions.</p>
<pre class="hljs"><code><div>ls
</div></code></pre>
<p>Output:</p>
<pre class="hljs"><code><div>02-assembly 03-mapping 04-outbreakSNP 05-strainCharacterization
</div></code></pre>
<p>Once our localization is correct we will launch nextflow with the next parameters:</p>
<ul>
<li>Raw reads</li>
<li>step outbreakMLST</li>
<li>gtf file needed for assembly step.</li>
</ul>
<pre class="hljs"><code><div>nextflow run ../../bacterial_wgs_training/main.nf \
    --reads <span class="hljs-string">'../RAW/DOWNSAMPLED/*R{1,2}*.fastq.gz'</span> \
    --fasta ../REFERENCES/listeria_NC_021827.1_NoPhagues.fna \
    --step outbreakMLST \
    --gtf ../REFERENCES/listeria_NC_021827.1_NoPhagues.gff \
    --scheme ../REFERENCES/MLST_listeria \
    --outdir 06-outbreakMLST \
    -resume \
    -profile conda
</div></code></pre>
<p>This will take a while as usual, and it is performed with a downsampled dataset, so we will describe here the results with the full dataset for practice our interpretation.</p>
<h3 id="results-analysis">Results analysis</h3>
<p>Let's proceed to analyze the results. We can find them in:</p>
<pre class="hljs"><code><div>/home/alumno/bacterial_wgs_training_dataset/RESULTS/06-outbreakMLST
</div></code></pre>
<p>Since alignment and quality control results has been previously addresed in this course (see <a href="03_QualityAndAssembly.md">02_QualityAndAssembly.md</a>, we will proceed to analyze cgMLST results.</p>
<p>The most important files at this point for cgMLST analsysis are <code>results.tsv</code> and <code>summary_result.tsv</code> files. Remaining files are useful for particular analysis where we may want to look at things not present at the cgMLST, or to explain some phenotipic behaviour.</p>
<p>We will focus on the main output in this exercise. In the summary file we will find which alleles have been found as exact match against a scheme allele, which ones were new inferred alleles, and which ones are alleles not found in or samples, have deletions/insertions or may be caused by a bad assembly.</p>
<p>In this case we obtain something like this:</p>
<div class="tables-start"></div>
<table>
<thead>
<tr>
<th>File</th>
<th>Exact match</th>
<th>INF</th>
<th>ASM_INSERT</th>
<th>ASM_DELETE</th>
<th>ALM_INSERT</th>
<th>ALM_DELETE</th>
<th>LNF</th>
<th>NIPH</th>
<th>NIPHEM</th>
<th>PLOT</th>
<th>ERROR</th>
</tr>
</thead>
<tbody>
<tr>
<td>RA-L2073</td>
<td>1744</td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2281</td>
<td>1747</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2327</td>
<td>1744</td>
<td>4</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2391</td>
<td>1747</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2450</td>
<td>1745</td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2677</td>
<td>1731</td>
<td>13</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>3</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2701</td>
<td>1740</td>
<td>8</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2709</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1748</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2782</td>
<td>1746</td>
<td>1</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2805</td>
<td>1745</td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>RA-L2978</td>
<td>1746</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>
<div class="tables-end"></div>
<p>But...it may be useful for you taking a look at the downsampling results this time, what happens with the cgMLST analysis when we use data with low coverage, and consequently a fragmented analysis? The summary results changes and we see this:</p>
<blockquote>
<p>PLOT alleles rise notably, this is because fragmented genome makes more probable the appearance of broken cds that fall in the start of end of a contig.</p>
</blockquote>
<h2 id="minimum-spanning-tree-visualization">Minimum spanning tree visualization</h2>
<p>In order to generate the minimum spanning tree from our <code>results.tsv</code> file we are going to use <a href="https://online.phyloviz.net/index">Phyloviz</a>, an online tool for MST visualization.</p>
<p>So..open click <a href="https://online.phyloviz.net/index">here</a> and phyloviz website should open</p>
<p align="center"><img src="img/phyloviz1.PNG" width="1000"></p>
<p>Next we need to upload our file, so we select profile data as input and select <em>Launch tree</em></p>
<p align="center"><img src="img/phyloviz2.PNG" width="1000"></p>
<p>We now have our minimum spanning tree but it looks pretty ugly and with little information. Let's add the samples names to the nodes. In order to do this we have to click on Graphic propertis in the left dropdown menu, click on nodes and check the Add Link Labels checkbox as shown in the image:</p>
<p align="center"><img src="img/phyloviz3.PNG" width="1000"></p>
<p>Next we are goint to add link labels which will show the absolute distance (number of alleles) among the nodes. As before we click on Graphic properties, next on Links and we check Add Link Labels checkbox.</p>
<p align="center"><img src="img/phyloviz4.PNG" width="1000"></p>
<p>Now we have a &quot;pretty&quot; minimum spanning tree with enough annotation for interpret our results. However we can also make it prettier (easy right?) adding some colors based on any locus of the profile or based on any auxiliary data we want to provide, p.e one useful data is the samples <strong>ST</strong>, for this we have to create</p>
<p align="center"><img src="img/phyloviz7.PNG" width="1000"></p>
<p>Finally we have out pretty MST, do you see any cluster? You can compare this result with the one obtained with the SNP-based pipeline.</p>
<p align="center"><img src="img/phyloviz5.PNG" width="1000"></p>
<h2 id="conclusion">Conclusion</h2>
<p>cgMLST and SNP-based approach generate the same result for this outbreak, we have between 0-5 different SNPs, and between 1-3 different alleles among the isolates belonging to the outbreak.</p>

</body>
</html>
